### PR TITLE
[SPARK-30539][PYTHON][SQL] Add DataFrame.tail in PySpark

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

https://github.com/apache/spark/pull/26809 added `Dataset.tail` API. It should be good to have it in PySpark API as well.

### Why are the changes needed?

To support consistent APIs.

### Does this PR introduce any user-facing change?

No. It adds a new API.

### How was this patch tested?

Manually tested and doctest was added.

### CODE REVIEW

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -605,6 +605,22 @@ def take(self, num):
         """
         return self.limit(num).collect()
 
+    @ignore_unicode_prefix
+    @since(3.0)
+    def tail(self, num):
+        """
+        Returns the last `num` rows in the Dataset.
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/dataframe.py]: Shall we match the format with line 601?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (only a minor comment on doc-string)

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Also, cc @BryanCutler .

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @HyukjinKwon .

**Comment by HyukjinKwon** [comment]
Thanks @dongjoon-hyun!

**Comment by zero323** [comment]
Just thinking out loud ‒ should there be `tailToPandas` variant? For `head` we can easily do `df.limit(n).toPandas()`.

**Comment by HyukjinKwon** [comment]
We could have things like `df.limit(n, reverse=True).toPandas()`; however, this implementation was added yet because we should think how it works with other SQL APIs. It looks pretty clear to have it in Scala/Python APIs; however, I wasn't sure about SQL.

**Pull request closed without merging**